### ISSUE TITLE
Cannot destroy newly created cluster 

### ISSUE THREAD

**Issue by ivelkov** [appcelerator-archive/amp#1445]
**Description**

Newly created amp cluster using amp version v0.11.0, build f7e0520e cannot be removed on Fedora 25

**Steps to reproduce the issue:**
1. amp -s localhost cluster create
```
[ivelin@ivelin-laptop amp-test] $ amp -s localhost cluster create
[localhost:50101]
wait for cluster
ok
cluster size is 1
deploy amp monitoring stack to cluster - stage 1
deploy ampmon.1.stack.yml => amp
Creating service amp_elasticsearch
Creating service amp_nats
Creating service amp_etcd
ok
wait for all amp monitoring stage 1 stack service replicas to be running
ok
deploy amp monitoring stack to cluster - stage 2
deploy ampmon.2.stack.yml => amp
Creating service amp_ampbeat
Creating service amp_kibana
ok
wait for all amp monitoring stage 2 service replicas to be running (540 sec)
ok
deploy portal stack to cluster
deploy portal.stack.yml => amp
Creating service amp_portal
ok
wait for all portal service replicas to be running (540 sec)
ok
deploy amp stack to cluster
deploy amp.stack.yml => amp
Creating service amp_agent
Creating service amp_amplifier
Creating service amp_amplifier-gateway
ok
wait for all amp service replicas to be running (540 sec)
ID                  NAME                    MODE                REPLICAS            IMAGE                                    PORTS
4tzu63t9wr2h        amp_ampbeat             replicated          1/1                 appcelerator/ampbeat:0.11.0              
5j4bx57d5uz5        amp_amplifier           global              1/1                 appcelerator/amplifier:0.11.0            *:50101->50101/tcp
7iuzrul84ds4        amp_nats                replicated          1/1                 appcelerator/amp-nats-streaming:v0.3.8   
bbzue61wmb9t        amp_etcd                replicated          3/3                 appcelerator/etcd:3.1.6                  
bzdiuzerqhwb        amp_kibana              replicated          1/1                 appcelerator/kibana:5.3.2                *:50106->5601/tcp
cyqj6mke6u68        amp_portal              replicated          1/1                 appcelerator/portal:0.11.0               
jpjvi89w8h0f        amp_agent               global              1/1                 appcelerator/agent:0.11.0                
lwnfa47udi3w        amp_elasticsearch       replicated          1/3                 appcelerator/elasticsearch-amp:5.3.2     
rhcac0c6j9ee        amp_amplifier-gateway   global              1/1                 appcelerator/amplifier-gateway:0.11.0    
ok
deploy amp proxy stack to cluster
deploy ampproxy.stack.yml => amp
Creating service amp_proxy
ok
wait for all amp proxy service replicas to be running (540 sec)

ok

Cluster status: healthy
[ivelin@ivelin-laptop amp-test] $ 
```
2. amp cluster ls
```
[ivelin@ivelin-laptop amp-test] $ amp -s localhost cluster ls
[localhost:50101]
f573e897-7aa0-4516-a195-42ee91039e97

```
3. amp cluster remove
```
[ivelin@ivelin-laptop amp-test] $ amp -s localhost cluster rm -s localhost:50101
[localhost:50101]
Error: No such object: infrakit
[ivelin@ivelin-laptop amp-test] $ echo $?
0
[ivelin@ivelin-laptop amp-test] $

```

**Describe the results you received:**

Cluster remove fails as shown by docker stack ls.
Exit code is 0.
```
[ivelin@ivelin-laptop amp-test] $ docker stack ls
NAME  SERVICES
amp   10
[ivelin@ivelin-laptop amp-test] $
```

**Describe the results you expected:**

1. Cluster remove doesn't fail
2. When cluster remove fails, exit code should not be 0

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
(paste your output here)
```

**Docker for Mac/Windows version, if applicable:**

```
[ivelin@ivelin-laptop amp-test] $ docker --version
Docker version 17.03.1-ce, build c6d412e
```

**Output of `amp version`:**

```
[ivelin@ivelin-laptop amp-test] $ amp --version
[cloud.appcelerator.io:50101]
amp version v0.11.0, build f7e0520e
[ivelin@ivelin-laptop amp-test] 
```

**Additional environment details (AWS, etc., if applicable):**

** Fedora 25 **
```
[ivelin@ivelin-laptop amp-test] $ cat /etc/*release*
Fedora release 25 (Twenty Five)
cat: /etc/lsb-release.d: Is a directory
NAME=Fedora
VERSION="25 (Workstation Edition)"
ID=fedora
VERSION_ID=25
PRETTY_NAME="Fedora 25 (Workstation Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:25"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=25
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=25
PRIVACY_POLICY_URL=https://fedoraproject.org/wiki/Legal:PrivacyPolicy
VARIANT="Workstation Edition"
VARIANT_ID=workstation
Fedora release 25 (Twenty Five)
Fedora release 25 (Twenty Five)
cpe:/o:fedoraproject:fedora:25
[ivelin@ivelin-laptop amp-test] $ 
```
** docker ps after cluster creation **
```
[ivelin@ivelin-laptop amp-test] $ docker ps
CONTAINER ID        IMAGE                                                                                                     COMMAND                  CREATED             STATUS                  PORTS                               NAMES
71de02da3a72        appcelerator/elasticsearch-amp@sha256:d37843bf2e96dd6c6f23ddade8917436cb3060bee49398b78487c7661f953001    "/bin/docker-entry..."   6 seconds ago       Up Less than a second   9200/tcp, 9300/tcp                  amp_elasticsearch.1.igqsrs3fwn4rq8150ol8rxi6a
6d7ffcb5d285        appcelerator/elasticsearch-amp@sha256:d37843bf2e96dd6c6f23ddade8917436cb3060bee49398b78487c7661f953001    "/bin/docker-entry..."   9 seconds ago       Up 2 seconds            9200/tcp, 9300/tcp                  amp_elasticsearch.3.4q8xfqxl0cumvv28nmgkumg24
c2cb5acc963d        dockercloud/haproxy@sha256:ad5c5d4802dee54d8ab5d3d51105d3facfa978eadc4dcced4c95658139efee5f               "/sbin/tini -- doc..."   7 minutes ago       Up 7 minutes            80/tcp, 443/tcp, 1936/tcp           amp_proxy.09zr3pew3me1r8am4d3r0ia1m.4w9bn5gihjc40rf8gahuii2qa
cdf115b2daba        appcelerator/amplifier-gateway@sha256:559d40a4df6784abb30f75956404d621cfa1fe14716417ed826fe620e3dcc730    "amplifier-gateway"      8 minutes ago       Up 8 minutes                                                amp_amplifier-gateway.09zr3pew3me1r8am4d3r0ia1m.l7j1x49r1ccmn1xie88n2rm1p
9d42590356c3        appcelerator/amplifier@sha256:074f9ea6ef6823c94b573edc6a9328575023596aca669c369e03632b9431a1de            "amplifier"              8 minutes ago       Up 8 minutes                                                amp_amplifier.09zr3pew3me1r8am4d3r0ia1m.iq3uku68q0xdkq138v4e7qora
caf3b80e974d        appcelerator/agent@sha256:ca6fc3df98af03adadcb0e16a0118ddb375c414a5a8eadaf9822fa005dc3fb15                "agent"                  8 minutes ago       Up 8 minutes                                                amp_agent.09zr3pew3me1r8am4d3r0ia1m.kjskq2bf59t649mi1bj36m1oj
87dff56f11ed        appcelerator/portal@sha256:60ff83bfa94f547b6bf82eb3f5f7c2d9b2e476caabb6ef50589f5d1b40f874be               "nginx -g 'daemon ..."   8 minutes ago       Up 8 minutes            80/tcp                              amp_portal.1.yryxrsretzega97d37x8ojda0
adc96a040b34        appcelerator/kibana@sha256:521433e5f5970ea2166c890649821ee633bc894c5a6fa829e01b772fea37fef2               "/bin/sh -c /run.sh"     8 minutes ago       Up 8 minutes            5601/tcp                            amp_kibana.1.y0388kblsncuz2wr7h66fhp15
869342f02839        appcelerator/ampbeat@sha256:c9ec2585b8ed7c8609f1dc9f0ee80bf244b078cd4d07e12c46b0cad0b480f64d              "/beat/ampbeat -e"       8 minutes ago       Up 8 minutes                                                amp_ampbeat.1.qr61sjpq5bqm21xpcrx3p6lsh
84fd3c2b8546        appcelerator/etcd@sha256:f83c7e4cbacb188703a79f1e8c47a597c2b905e8fd9149eede8a5062aff3ab1e                 "/bin/run.sh --adv..."   8 minutes ago       Up 8 minutes            2379-2380/tcp, 4001/tcp, 7001/tcp   amp_etcd.2.qlb3py42vmd0v83mybc5o82b6
083995fdb8d7        appcelerator/etcd@sha256:f83c7e4cbacb188703a79f1e8c47a597c2b905e8fd9149eede8a5062aff3ab1e                 "/bin/run.sh --adv..."   8 minutes ago       Up 8 minutes            2379-2380/tcp, 4001/tcp, 7001/tcp   amp_etcd.3.1aj538i9414359j8yb00eacsb
d4090e0846df        appcelerator/etcd@sha256:f83c7e4cbacb188703a79f1e8c47a597c2b905e8fd9149eede8a5062aff3ab1e                 "/bin/run.sh --adv..."   8 minutes ago       Up 8 minutes            2379-2380/tcp, 4001/tcp, 7001/tcp   amp_etcd.1.qammgcqfz6h8z13ccn25ifdhy
8b32e3451ae4        appcelerator/amp-nats-streaming@sha256:95a5df8fb61a8dd2f856be2e9584c1fb270f0abd72dff9ee7ea2f5fde7b1ea3a   "/nats-streaming-s..."   8 minutes ago       Up 8 minutes            4222/tcp, 6222/tcp, 8222/tcp        amp_nats.1.vqhqql7tyf20g9tkdb9qn2jm4
[ivelin@ivelin-laptop amp-test] $ 
```

**Comment by JosephGJ** [comment]
related to #1212 and #1402 

For the next release, this will be fully fixed. For now, you can remove the amp local cluster with
```
$ docker stack rm amp
```

**Comment by ivelkov** [comment]
Cool, just to note that there are two aspects of the issue - one being the actual issue and the other one the exit code of the CLI.
IMHO the CLI exit code should not be 0 in case of failures.

**Comment by JosephGJ** [comment]
Fixed in 0.12.0 release
